library/alloc/benches/btree/map.rs

```diff
@@ -290,7 +290,7 @@ where
         let mut c = 0;
         for i in 0..BENCH_RANGE_SIZE {
             for j in i + 1..BENCH_RANGE_SIZE {
-                black_box(map.range(f(i, j)));
+                let _ = black_box(map.range(f(i, j)));
```

Comment:
Please don't add `black_box`; as this shows, it's legitimate to call it on a computation you want to force but whose value you don't care about.

Reply:
Wait, scratch that. Is must_use checking somehow seeing *through* `black_box` here to `map.range`?

Reply:
Somehow it did, yes. Now that you mention it... huh? See the [rust-log-analyzer output](https://github.com/rust-lang/rust/pull/89899#issuecomment-943853776) from above.